Add default value for `$numberOfBacktraces` in `SettledResult::throw`

inxilpro opened the pull request:
#131 changed the signature of `SettledResult::throw` so that calling `->throw()` without a parameter is no longer allowed. This just sets the default to `NULL` so that it's backwards-compatible.

inxilpro committed:
Fix code styling [ci skip]

inxilpro committed:
Set default for `$numberOfBacktraces`

daniel-g-wood commented:
Hey @inxilpro , unfortunately this will still fail in some cases. If you pass `null` to this function and there are errors, it fails on `$this->errorAsString($numberOfBacktraces)`. `errorAsString` requires an int parameter, so it throws a type error there instead. You'll also need make the following change (or something like this):
```
    public function errorAsString(?int $numberOfBacktraces = 2)
    {
        $numberOfBacktraces ??= 2;
        ....
    }
```

inxilpro committed:
Set default backtrace number in errorAsString call

Ensures that the errorAsString method receives a default value of 2 for numberOfBacktraces if none is provided, preventing potential null value issues.

inxilpro committed:
Fix code styling [ci skip]

inxilpro committed:
Merge remote-tracking branch 'upstream/main' into settledresult-throw-bc-fix
